.github/workflows/node-ci.yml

```diff
@@ -0,0 +1,136 @@
+name: node-ci
+
+on:
+  workflow_dispatch:
+  push:
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - "platform/android/**"
+      - "platform/ios/**"
+      - "platform/qt/**"
+      - ".gitignore"
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - "platform/android/**"
+      - "platform/ios/**"
+      - ".gitignore"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os:
+          - macos-11.0
+          - macos-10.15
+          - ubuntu-20.04
+          - ubuntu-18.04
+    runs-on: ${{ matrix.os }}
+    env:
+      BUILDTYPE: ${{github.ref == 'refs/heads/master' && 'Release' || 'Debug'}}
+
+    defaults:
+      run:
+        working-directory: ./
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Install dependencies
+        if: runner.os == 'macOS'
+        env:
+          HOMEBREW_NO_AUTO_UPDATE: 1
+          HOMEBREW_NO_INSTALL_CLEANUP: 1
+        run: |
+          brew list cmake || brew install cmake
```

Comment:
Here and for Linux environment, similar to suggestions below, suggest consolidating this to the minimal set of dependencies required to build on Github.  Here, this should include only the strictly required deps and not development or optional deps.